anthony-timea/scripts/articleView.js

```diff
@@ -0,0 +1,113 @@
+let articleView = {};
+
+articleView.populateFilters = function() {
+  $('article').each(function() {
+    // REVIEW: We can declare several variables at once and assign their values later when using let. Keep in mind that we cannot do this with const.
+    let authorName, category, optionTag;
+    if (!$(this).hasClass('template')) {
+      // REVIEW: We need to take every author name from the page, and make it an option in the Author filter.
+      // To do so, Build an <option> DOM element that we can append to the author <select> element.
+      // $(this).hasClass('data-author').text(optionTag);
+      // Start by grabbing the author's name from `this` article element, and then use that bit of text to create the option tag (in a variable named `optionTag`) that we can append to the #author-filter select element.
+      authorName = $(this).attr('data-author');
+
+      // optionTag = '<option value="' + authorName + '">' + authorName + '</option>';
+      optionTag = `<option value="${authorName}">${authorName}</option>`;
+
+      if ($('#author-filter option[value="' + authorName + '"]').length === 0) {
+        $('#author-filter').append(optionTag);
+      }
+
+      // REVIEW: Similar to the above, but...
+      // Avoid duplicates! We don't want to append the category name if the <select> already has this category as an option!
+      category = $(this).attr('data-category');
+
+      // Done: Refactor this concatenation using a template literal.
+      // optionTag = '<option value="' + category + '">' + category + '</option>';
+      optionTag = `<option value="${category}">${category}</option>`;
+
+      if ($('#category-filter option[value="' + category + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    // REVIEW: Inside this function, "this" is the element that triggered the event handler function we are defining. "$(this)" is using jQuery to select that element (analogous to event.target that we have seen before), so we can chain jQuery methods onto it.
+    if ($(this).val()) {
+      // DONE: If the <select> menu was changed to an option that has a value, we first need to hide all the articles, and then show just the ones that match for the author that was selected.
+      $('article').hide();
+      // !$(`article[data-author="${$(this).val()}"]`).hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn(200);
+      console.log($(this).val());
+      // Use an "attribute selector" to find those articles, and fade them in for the reader.
+
+    } else {
+      // DONE: If the <select> menu was changed to an option that is blank, we should first show all the articles, except the one article we are using as a template.
+      $('article').show();
+    }
+
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
```

Comment:
you need to make sure that when you click back to the default setting it shows all the articles again